### PR TITLE
#162725255  Receive email on successful signup

### PR DESCRIPTION
**Pull request by MuhweziDeo**
#### What does this PR do?
implement receive email on successful signup

#### Description of Task to be completed?
- send an email to  a user's email they used on signup 
- verify the email once a user clicks on the verification link sent to their inbox

#### How should this be manually tested?
- fetch this branch by running the command git fetch origin `ft-registration-emial-verification-162725255` in the terminal.
- check into the branch `git checkout ft-registration-emial-verification-162725255`
- add **EMAIL_HOST_USER** and **EMAIL_HOST_PASSWORD** to your .env file
- start the virtual environment `source venv/bin/activate`
- migrate to update database  using `python manage.py migrate`
-  run development server using `python manage.py runserver`
- using postman perform a post request on `http://localhost:8000/api/users/`  to register a user
- provide data in format
        { 
           "user":{
    
        "password":"randompassworD1",
        "email":"radomemail@gmail.com",
        "username":"deoaggrey"
       }     } 
- ensure to enter a valid email 
- check the email you used to signedup with for a link in the format `localhost:8000/api/users/registration-confirmation/MzQ/52z-49bc0b91cdc08b29f3ec/`
- perform a get request on the link and a response will be returned to confirm successful email verification

- run `coverage run --omit="*tests*,*init*,*migrations*,*wsgi*" --source="authors" manage.py test&&coverage report -m` in terminal to test locally

#### What are the relevant pivotal tracker stories?
[#162725255](https://www.pivotaltracker.com/story/show/162725255)

### Screenshots
screen shot 2019-01-14 at 2 41 56 pm

### CODE REVIEW

**Review by kdaglas — CHANGES_REQUESTED** [review]
the PR title is not following our PR conventions, maybe fix that as i test the branch

**Review by malfahad — COMMENTED** [review]
This commit seems to be missing a file. Can't find the utils file with the 'email verification token creator' class.

**Comment by ja-odur** [comment]
@MuhweziDeo Since you did not add any dependency to the `requirement.txt` file, please dont include installing requirements in the `testing section.

**Comment by ja-odur** [comment]
@MuhweziDeo You also do not need to makemigrations since the migrations are pushed to the `github`. you only need to migrate.

**Comment by ja-odur** [comment]
@MuhweziDeo Please rename the test folders from `test_` to `test`

**Comment by ja-odur** [comment]
@MuhweziDeo the screenshots in the PR should be in the `screenshots` section. please check the engineering playbook.

File: authors/apps/authentication/tests_/test_endpoints.py [hunk 1]
```diff
@@ -5,14 +5,57 @@
...
 
 class TestUserEndPoints(BaseTestClass):
     def test_user_regsitration_with_valid_data_succeeds(self):
         resp = self.client.post(reverse('authentication:signup'),
                                 content_type='application/json', data=json.dumps(self.user_data))
         self.assertEqual(resp.status_code, status.HTTP_201_CREATED)
-        self.assertIn('Jacob', str(resp.data))
+        self.assertIn(self.user_data['user']['username'], str(resp.data))
+
+    def test_user_email_verification_email_sent(self):
```
Comment (ja-odur) [inline authors/apps/authentication/tests_/test_endpoints.py]: @MuhweziDeo Please use the test `succeeds` convention instead of `sent`.

**Comment by MuhweziDeo** [comment]
> @MuhweziDeo Since you did not add any dependency to the `requirement.txt` file, please dont include installing requirements in the `testing section.

@ja-odur  my bad was reusing PR message

**Comment by MuhweziDeo** [comment]
> @MuhweziDeo the screenshots in the PR should be in the `screenshots` section. please check the engineering playbook.

@ja-odur  noted

**Comment by MuhweziDeo** [comment]
> @MuhweziDeo Please rename the test folders from `test_` to `test`

@ja-odur  this no implication in anyway but will do

**Comment by MuhweziDeo** [comment]
> This commit seems to be missing a file. Can't find the utils file with the 'email verification token creator' class.

@malfahad  sure

**Review by kdaglas — DISMISSED** [review]

**Comment by kdaglas** [comment]
Can also fix the spelling errors in your PR, for example, that email spelling and also start with a capital letter on the PR title

**Comment by mariamiah** [comment]
screen shot 2019-01-16 at 4 01 51 pm
Yes the code works well, though update the PR, and change this issue concerning the console, to direct check in the email

**Review by calvinpete — DISMISSED** [review]
Looks Good To Me

**Review by d-rita — DISMISSED** [review]
works fine. though be mindful of typos when creating the branch in future.

File: authors/apps/authentication/tests_/test_model.py [hunk 2]
```diff
@@ -3,32 +3,91 @@
...
         with self.assertRaises(TypeError):
             user = User.objects.create_user(
                 username=None, email='aggrey90@gmail.com', password='randompassword')
 
     def test_return_str__method(self):
+        """
+        Test __str__ method on user model
+
+        Returns:
+            when __str__ is callted is should return a users email
```
Comment (ja-odur) [inline authors/apps/authentication/tests_/test_model.py]: @MuhweziDeo There is a typo here.

**Comment by MuhweziDeo** [comment]
> works fine. though be mindful of typos when creating the branch in future.

@d-rita  my apologies to you and the team for that

File: authors/apps/authentication/tests_/test_model.py [hunk 3]
```diff
@@ -3,32 +3,91 @@
...
         self.user = User.objects.create_user(
             username="aggrey", email="aggrey256@gmail.com", password='randompassword')
         self.assertEqual(self.user.__str__(), 'aggrey256@gmail.com')
 
     def test_return_short_name__method(self):
+        """
+        Test get_short_name method on user model
+
+        Returns:
+            when get_short_name is callted is should return a users username
```
Comment (ja-odur) [inline authors/apps/authentication/tests_/test_model.py]: @MuhweziDeo typo here too.

**Review by kdaglas — DISMISSED** [review]
ok done testing, the email is sent to show successful registration that is dope, tests are on point... LGTM

**Review by mariamiah — DISMISSED** [review]
LGTM

File: authors/apps/authentication/tests_/test_model.py (hunk 2 above)
Comment (kdaglas) [reply authors/apps/authentication/tests_/test_model.py]: I had not seen that... thanx @ja-odur

**Review by kdaglas — APPROVED** [review]

**Review by ja-odur — APPROVED** [review]
LGTM

**Review by d-rita — APPROVED** [review]

**Review by kdaglas — COMMENTED** [review]
LGTM

**Pull request merged**